test/e2e/framework/framework.go

```diff
@@ -352,6 +352,21 @@ func UpdateDeployment(kubeClientSet kubernetes.Interface, namespace string, name
 	return nil
 }
 
+// UpdateIngress runs the given updateFunc on the ingress and waits for it to be updated
+func UpdateIngress(kubeClientSet kubernetes.Interface, namespace string, name string, updateFunc func(d *extensions.Ingress) error) error {
+	ingress, err := kubeClientSet.ExtensionsV1beta1().Ingresses(namespace).Get(name, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	if err := updateFunc(ingress); err != nil {
+		return err
+	}
+
+	_, err = kubeClientSet.ExtensionsV1beta1().Ingresses(namespace).Update(ingress)
```

Comment:
> waits for it to be updated

I don't think this is blocking. Does the function really waits for it to be updated? Not saying make it wait, but maybe fix the comment.

Reply:
true, updated.